models/Challenge.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+const queries = require('./queries');
+const utils = require('../utils');
+
+class Challenge {
+  makeChallenge(userId, spotId) {
+    const sql = queries.makeChallenge;
+
+    return new Promise((resolve, reject) => {
+      const client = utils.db.createClient();
+
+      client.connect();
+      client.on('error', (err) => { reject(err); });
+
+      const query = client.query(sql, [userId, spotId]);
+
+      query.on('row', (result) => {
+        client.end();
+        resolve(result);
+      });
+
+      query.on('error', (err) => {
+        client.end();
+        reject({ error: err });
```

Comment:
In some places, we're rejecting with `reject({ error: err })` in others we're just calling `reject(err)`. Pick one and be consistent